docs/guide/resources/synthetic_dataset_generation.md

```diff
@@ -0,0 +1,65 @@
+Few shot learning is one of the most powerful tools that autolabel offers to improve the accuracy of LLM generated labels. However, curating a seed dataset to use for few shot learning can be a time consuming and tedious process. To make this process easier, autolabel's LabelingAgent provides a method to generate synthetic datasets. These datasets can be used as seed datasets for few shot learning or for any other purpose. This guide will walk you through the process of generating a synthetic dataset using autolabel.
+
+Currently, autolabel supports synthetic dataset generation for classification and entity matching tasks. We plan to add support for other task types in the future.
+
+### **Walkthrough: Creating a Synthetic Dataset for Banking**
+
+<ol>
+<li>The first step is to import the LabelingAgent from autolabel. This is the main class that we will use to generate the synthetic dataset.
+
+```python
+from autolabel import LabelingAgent
+```
+
+</li>
+<li>The next step is to create the task config. Make sure to add the <code>dataset_generation</code> section to the config. This section contains the parameters for the dataset generation process. The <code>guidelines</code> parameter is a string containing the guidelines for the dataset generation task. The <code>num_rows</code> parameter is an integer indicating the number of rows <em><strong>per label</strong></em> to generate in the dataset.
+
+```python
+config = {
+  "task_name": "BankingComplaintsClassification",
+  "task_type": "classification",
+  "dataset": {
+    "label_column": "label",
+    "delimiter": ","
+  },
+  "model": {
+    "provider": "openai",
+    "name": "gpt-3.5-turbo"
+  },
+  "prompt": {
+    "task_guidelines": "You are an expert at understanding bank customers support complaints and queries.\nYour job is to correctly classify the provided input example into one of the following categories.\nCategories:\n{labels}",
+    "output_guidelines": "You will answer with just the the correct output label and nothing else.",
+    "labels": [
+      "activate_my_card",
+      ..., # other labels
+      "wrong_exchange_rate_for_cash_withdrawal"
+    ],
+    "example_template": "Input: {example}\nOutput: {label}"
+  },
+  "dataset_generation": {
+    "num_rows": 5,
+    "guidelines": "You are an expert at generating synthetic data. You will generate a dataset that satisfies the following criteria:\n1. The data should be diverse and cover a wide range of scenarios.\n2. The data should be as realistic as possible, closely mimicking real-world data.\n3. The data should vary in length, some shorter and some longer.\n4. The data should be generated in a csv format.\n\nEach row should contain a realistic bank complaint. Use CSV format, with each line containing just the complaint and nothing else."
```

Comment:
Do we give the model an example input? eg. few shot?

Reply:
I thought we decided to not use few shot because if they had those examples already, they wouldn't need to generate more. I can add it back in a following PR though!

Reply:
hmm makes sense. I was under the impression we send some example input to get good results. If that's not true, no few shot is also fine. However, in that case, how do we specify the type of each input and what each input means?

Reply:
update: read the answer for the last question, looks like it is going to be added to the description. Sounds good in that case